internal/publish/pull_requests.go

```diff
@@ -0,0 +1,116 @@
+	expectedTitle := buildPullRequestTitle(manifest)
+	q := fmt.Sprintf(`repo:elastic/package-storage base:snapshot is:pr is:open in:title "%s"`, expectedTitle)
+	logger.Debugf("Use Search API to find an open pull request (query: '%s')", q)
+	searchResults, _, err := githubClient.Search.Issues(context.Background(), q, new(github.SearchOptions))
+	if err != nil {
+		return false, errors.Wrap(err, "can't search pull requests")
+	}
+
+	for _, item := range searchResults.Issues {
+		if *item.Title == expectedTitle {
+			logger.Debugf("Found pull request: ", *item.HTMLURL)
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func openPullRequest(githubClient *github.Client, githubUser, destinationBranch string, manifest packages.PackageManifest, commitHash string) error {
+	title := buildPullRequestTitle(manifest)
+	diffURL := buildPullRequestDiffURL(githubUser, commitHash)
+	description := buildPullRequestDescription(manifest, diffURL)
+
+	userHead := fmt.Sprintf("%s:%s", githubUser, destinationBranch)
+	maintainerCanModify := true
+	base := snapshotStage
+
+	logger.Debugf("Create new pull request (head: %s, base: %s)", userHead, base)
+	pullRequest, _, err := githubClient.PullRequests.Create(context.Background(), repositoryOwner, repositoryName, &github.NewPullRequest{
+		Title:               &title,
+		Head:                &userHead,
+		Base:                &base,
+		Body:                &description,
+		MaintainerCanModify: &maintainerCanModify,
+	})
+	if err != nil {
+		return errors.Wrap(err, "can't open new pull request")
+	}
+
+	logger.Debugf("Pull request URL: %s", *pullRequest.HTMLURL)
+
+	// Try to set reviewers
+	reviewers := buildReviewersRequest(manifest)
+	logger.Debugf("Update reviewers (pull request ID: %d)", *pullRequest.Number)
+	pullRequest, _, err = githubClient.PullRequests.RequestReviewers(context.Background(), repositoryOwner, repositoryName, *pullRequest.Number, reviewers)
+	if err != nil {
+		return errors.Wrap(err, "can't request reviewers, please double-check if the owner exists and has access to the repository")
+	}
+
+	if len(pullRequest.RequestedTeams) != 0 || len(pullRequest.RequestedReviewers) != 0 {
+		logger.Debugf("Reviewers requested successfully (teams: %d, reviewers: %d)", len(pullRequest.RequestedTeams), len(pullRequest.RequestedReviewers))
+		return nil
+	}
+
+	// Fallback reviewers to default package owner
+	logger.Debugf("Update reviewers with default owner (pull request ID: %d)", *pullRequest.Number)
+	pullRequest, _, err = githubClient.PullRequests.RequestReviewers(context.Background(), repositoryOwner, repositoryName, *pullRequest.Number,
+		buildDefaultReviewersRequest())
+	if err != nil {
+		return errors.Wrap(err, "can't request reviewers with default owner")
+	}
+	return nil
+}
+
+func buildPullRequestTitle(manifest packages.PackageManifest) string {
+	return fmt.Sprintf(`[snapshot] Update "%s" integration to version %s`, manifest.Name, manifest.Version)
```

Comment:
It might not be an integration package.

```suggestion
	return fmt.Sprintf(`[snapshot] Update "%s" package to version %s`, manifest.Name, manifest.Version)
```